dll/win32/shell32/shellrecyclebin/recyclebin_v5_enumerator.c

```diff
@@ -34,6 +41,14 @@ RecycleBin5File_RecycleBinFile_QueryInterface(
         *ppvObject = &s->recycleBinFileImpl;
     else if (IsEqualIID(riid, &IID_IRecycleBinFile))
         *ppvObject = &s->recycleBinFileImpl;
+    else if (IsEqualIID(riid, &IID_IExtractIconA) || IsEqualIID(riid, &IID_IExtractIconW))
+    {
+        DWORD dwAttributes;
+        if (RecycleBin5File_RecycleBinFile_GetAttributes(This, &dwAttributes) == S_OK)
+            return SHCreateFileExtractIconW(s->FullName, (dwAttributes & FILE_ATTRIBUTE_DIRECTORY) ? FILE_ATTRIBUTE_DIRECTORY : FILE_ATTRIBUTE_NORMAL, riid, ppvObject);
```

Comment:
Any reason we can't just pass in `dwAttributes` as-is?

Reply:
indeed here there's no need to work around the stuff

Reply:
oh, it's true, it can be used directly